components/higher-order/with-filters/index.js

```diff
@@ -1,25 +1,63 @@
+/**
+ * External dependencies
+ */
+import { debounce, uniqueId } from 'lodash';
+
 /**
  * WordPress dependencies
  */
 import { Component, getWrapperDisplayName } from '@wordpress/element';
-import { applyFilters } from '@wordpress/hooks';
+import { addAction, applyFilters, removeAction } from '@wordpress/hooks';
+
+const ANIMATION_FRAME_PERIOD = 16;
```

Comment:
Why not use `requestAnimationFrame` ?

https://caniuse.com/#feat=requestanimationframe

Reply:
I will check if it works the same way as debounce 👍

Reply:
We polyfill `requestAnimationFrame` in our test setup which makes the tests I added fail :)

Should I polyfill `requestAnimationFrame` using `debounce` in Jest config? :)

Reply:
It turns out `requestAnimationFrame` works exactly like `setTimeout( callback, computedMillisecondsUntilNextAnimationFrame );`, which isn't exactly what we need here. When you call:
```js
requestAnimationFrame( () => console.log( 'test 1' ) ); requestAnimationFrame( () => console.log( 'test 2' ) ); const x = requestAnimationFrame( () => console.log( 'test 3' ) );
```
It will log all 3 values:
```bash
test 1
test 2
test 3
```

We want to have the following when calling:
```js
const log = _.debounce( () => console.log( 'test' ), 16 );
log(); log(); log();
```

```bash
test
```

Reply:
Ah, yes, that makes sense. Seems reasonable as it is.